docs/guide/transformers.md

```diff
@@ -1,7 +1,295 @@
+    from pprint import pprint
+    from langchain_graph_retriever.transformers.spacy import SpacyNERTransformer
+    spacy = SpacyNERTransformer()
+
+    spacy_docs = spacy.transform_documents(docs)
+    for doc in spacy_docs:
+        pprint({"id": doc.id, "metadata": doc.metadata}, width=100)
+    ```
+
+Example output:
+    ```text
+    {'id': 'red_fox',
+    'metadata': {'entities': ['ORG: The National Wildlife Federation',
+                            'GPE: New York City',
+                            'ORG: the Wildlife Conservation Summit',
+                            'DATE: 2023']}}
+    {'id': 'sea_turtle',
+    'metadata': {'entities': ['ORG: The World Wildlife Fund',
+                            'FAC: the Great Barrier Reef',
+                            'ORG: the Marine Life Protection Conference',
+                            'LOC: The Green Sea Turtle',
+                            'DATE: 2024']}}
+    ```
+
+### Parser Based
+
+The following document transformer uses a parser to extract metadata.
+
+#### HyperlinkTransformer
+
+The [`HyperlinkTransformer`][langchain_graph_retriever.transformers.html.HyperlinkTransformer]
+extracts hyperlinks from HTML content and stores them in document metadata.
+
+??? example "Test Html Documents"
+    ```python
+    from langchain_core.documents import Document
+    animal_html = """
+        <!DOCTYPE html>
+        <html><head><title>Animals of the World</title></head>
+        <body>
+            <h2>Mammals</h2>
+            <p>The <a href="https://example.com/lion">lion</a> is the king of the jungle.</p>
+            <p>The <a href="https://example.com/elephant">elephant</a> is a large animal.</p>
+
+            <h2>Birds</h2>
+            <p>The <a href="https://example.com/eagle">eagle</a> soars high in the sky.</p>
+            <p>The <a href="https://example.com/penguin">penguin</a> thrives in icy areas.</p>
+        </body></html>
+        """
+
+    html_doc = Document(
+        page_content=animal_html,
+        metadata={"url": "https://example.com/animals"}
+    )
+    ```
+
+    Note that each document needs to have an existing `url` metadata field.
+
+Example use:
+    ```python
+    from pprint import pprint
+    from langchain_graph_retriever.transformers.html import HyperlinkTransformer
+    html_transformer = HyperlinkTransformer()
+
+    extracted_doc = html_transformer.transform_documents(html_docs)[0]
+
+    pprint(extracted_doc.metadata)
+    ```
+
+Example output:
+    ```text
+    {'hyperlink': ['https://example.com/eagle',
+                'https://example.com/lion',
+                'https://example.com/elephant',
+                'https://example.com/penguin'],
+    'url': 'https://example.com/animals'}
+    ```
+
+## Metadata Utilities
+
+### ParentTransformer
+
+The [`ParentTransformer`][langchain_graph_retriever.transformers.ParentTransformer]
+adds the hierarchal `Parent` path to the document metadata.
+
+??? example "Test Documents"
+    ```python
```

Comment:
I bet these could be snippets from the existing tests!

Reply:
yes, would be amaze!